output/tar/tar_output.go

```diff
@@ -51,10 +51,9 @@ func (i Output) Apply(rootPath string) <-chan error {
 
 		err = tar.Run()
 		if err != nil {
-			done <- err
+			done <- output.Report{output.UnknownError.Wrap(err).(*errors.Error), def.Output{}}
 			return
 		}
-
 	}()
 	return done
```

Comment:
Yep, since it's a value channel (rather than channeling a pointer), nil is not possible, and a read returning from a closed channel will use the "zero" value: http://golang.org/ref/spec#Receive_operator  So there will be no panic on those checks.

Question: *Should* the error cases also be required to return the filled-in output?  Could go either way; by comparison, if this was a multi-return method and could produce a `*def.Output`, we'd surely return nil.

Trying that extra function with a return value, but it came out pretty ugly and made me notice some other incongruities with our error handling.  We have errors being sent directly into the report channel that don't pass through the type filtering the catch blocks, and that seems wrong.  So I think I'm going to switch everything to panics inside the `try` block, and that kills both birds with one stone... except for making sure there's a success return, but that should consistently be a `done <- output.Report{nil, o.spec}` in the last line, and I can't actually figure out a way to make that less copypasta.  (...Maybe we should add a `Success`-only handler to `errors/try`? o.o  But naw, that's isomorphic to the last lines of the try block no matter what.)